test/unit/request-wrapper.test.js

```diff
@@ -680,3 +718,156 @@ describe('formatError', () => {
     expect(error.message).toBe('error in building the request');
   });
 });
+
+describe('gzipRequestBody', () => {
+  const gzipSpy = jest.spyOn(zlib, 'gzipSync');
+  const errorLogSpy = jest.spyOn(logger, 'error').mockImplementation(() => {});
+  const debugLogSpy = jest.spyOn(logger, 'debug').mockImplementation(() => {});
+
+  afterEach(() => {
+    gzipSpy.mockClear();
+    errorLogSpy.mockClear();
+    debugLogSpy.mockClear();
+  });
+
+  it('should return unaltered data if encoding header is already set to gzip', async () => {
+    let data = 42;
+    const headers = { 'Content-Encoding': 'gzip' };
+
+    data = await requestWrapperInstance.gzipRequestBody(data, headers);
+    expect(data).toBe(42);
+  });
+
+  it('should return unaltered data if encoding header values include gzip', async () => {
+    let data = 42;
+    const headers = { 'Content-Encoding': ['gzip', 'other-value'] };
+
+    data = await requestWrapperInstance.gzipRequestBody(data, headers);
+    expect(data).toBe(42);
+  });
+
+  it('should return unaltered data if data is null', async () => {
+    let data = null;
+    const headers = {};
+
+    data = await requestWrapperInstance.gzipRequestBody(data, headers);
+    expect(gzipSpy).not.toHaveBeenCalled();
+    expect(data).toBe(null);
+  });
+
+  it('should compress json data into a buffer', async () => {
+    let data = { key: 'value' };
+    const headers = {};
+
+    const jsonSpy = jest.spyOn(JSON, 'stringify');
+
+    data = await requestWrapperInstance.gzipRequestBody(data, headers);
```

Comment:
We also should verify that Content-Encoding was set to "gzip".  There are a few other tests where this check should be added as well.

Reply:
I am checking that explicitly in its own test [here](https://github.com/IBM/node-sdk-core/pull/111/files#diff-5857e7caf9c146751fb8caef9bae6024R813) but I'm happy to add that check to the other tests as well if you think I should.

Reply:
I went ahead and added that check to all gzip tests 👍